keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/vetos/RequiredLoadBalancerVeto.kt

```diff
@@ -0,0 +1,95 @@
+package com.netflix.spinnaker.keel.ec2.vetos
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.OverrideableClusterDependencyContainer
+import com.netflix.spinnaker.keel.api.ec2.loadBalancersByRegion
+import com.netflix.spinnaker.keel.api.ec2.targetGroupsByRegion
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.clouddriver.model.AmazonLoadBalancer
+import com.netflix.spinnaker.keel.clouddriver.model.ApplicationLoadBalancerModel
+import com.netflix.spinnaker.keel.clouddriver.model.ClassicLoadBalancerModel
+import com.netflix.spinnaker.keel.core.api.DEFAULT_SERVICE_ACCOUNT
+import com.netflix.spinnaker.keel.veto.Veto
+import com.netflix.spinnaker.keel.veto.VetoResponse
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.withContext
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class RequiredLoadBalancerVeto(
+  private val cloudDriver: CloudDriverService
+) : Veto {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override suspend fun check(resource: Resource<*>): VetoResponse {
+    val spec = resource.spec as? OverrideableClusterDependencyContainer<*>
+      ?: return allowedResponse()
+
+    // avoid making call to CloudDriver to get all the load balancers if we don't need to
+    if (spec.loadBalancersByRegion.isEmpty() && spec.targetGroupsByRegion.isEmpty()) {
+      return allowedResponse()
+    }
+
+    val loadBalancers = loadBalancers(spec)
+    val missingLoadBalancers = loadBalancers.checkLoadBalancers(spec)
+
+    return if (missingLoadBalancers.isEmpty()) {
+      allowedResponse()
+    } else {
+      deniedResponse(
+        message = missingLoadBalancers.joinToString(separator = "\n", transform = MissingDependency::message),
+        vetoArtifact = false
+      )
+    }
+  }
+
+  private suspend fun loadBalancers(spec: OverrideableClusterDependencyContainer<*>) =
+    withContext(IO) {
```

Comment:
This is the first time I've seen the `IO` context used in keel. Should we be using it in more places?

Reply:
My understanding is that any time we're doing blocking IO (as in here, where I'm calling CloudDriver), that's the appropriate one to use.

Reply:
Reference(s), please?

Reply:
Javadoc on `IO`